libqtile/backend/wayland/qw/keyboard.c

```diff
@@ -39,20 +64,52 @@ static void qw_keyboard_handle_key(struct wl_listener *listener, void *data) {
     // Get current keyboard modifiers (shift, ctrl, alt, etc.)
     uint32_t modifiers = wlr_keyboard_get_modifiers(keyboard->wlr_keyboard);
 
-    // If key is pressed, call user callback for each symbol to check if handled
+    // If key is pressed...
     if (event->state == WL_KEYBOARD_KEY_STATE_PRESSED) {
+        // Track key for repeat
+        keyboard->key_pressed = true;
+
+        // Call user callback for each symbol to check if handled
         for (int i = 0; i < nsyms; ++i) {
             // TODO: for efficiency maybe let c take control of the key list?
             // If callback returns 1, event is handled; no further processing needed
             if (server->keyboard_key_cb(syms[i], modifiers, server->cb_data) == 1) {
                 handled = true;
+                keyboard->repeat_keysym = syms[i];
                 break;
             }
         }
-    }
 
-    // If not handled, forward the key event to the seat for default processing
-    if (!handled) {
+        if (handled) {
+            // Set up timer to repeat key press
+            if (!keyboard->repeat_source) {
```

Comment:
== NULL

Reply:
Sorry - just very used to not doing these explicit checks in python.